nav2_mppi_controller/include/nav2_mppi_controller/tools/utils.hpp

```diff
@@ -510,8 +510,71 @@ inline void savitskyGolayFilter(
     control_sequence.wz(offset)};
 }
 
+/**
+ * @brief Find the first pose at which there is an in-place rotation in the path
+ * @param path Path to check
+ * @param translation_threshold Maximum translation per pose transition
+ * @param rotation_threshold Minimum total rotation to consider significant
+ * @return Index after the rotation sequence if found, path size if no rotation detected
+ */
+inline unsigned int findFirstPathRotation(
+  const nav_msgs::msg::Path & path,
+  float translation_threshold,
+  float rotation_threshold)
+{
+  if (path.poses.size() < 2) {
+    return path.poses.size();
+  }
+
+  // Iterate through path to find first rotation sequence
+  for (unsigned int start_idx = 0; start_idx < path.poses.size() - 1; ++start_idx) {
+    // Check initial translation
+    float ab_x = path.poses[start_idx + 1].pose.position.x -
+      path.poses[start_idx].pose.position.x;
+    float ab_y = path.poses[start_idx + 1].pose.position.y -
+      path.poses[start_idx].pose.position.y;
+    float initial_translation = sqrtf(ab_x * ab_x + ab_y * ab_y);
+
+    if (initial_translation >= translation_threshold) {
+      continue;  // Not a rotation sequence, try next pose
+    }
+
+    // Start of potential rotation sequence
+    float start_yaw = tf2::getYaw(path.poses[start_idx].pose.orientation);
```

Comment:
Can't we just compare the heading if they're within `1e-3` of each other if they're different. If so, that's an in-place rotation? It really seems like this should be much simpler.